src/System.Linq.Queryable/src/System/Linq/EnumerableRewriter.cs

```diff
@@ -120,15 +120,13 @@ private static Type GetPublicType(Type t)
             // we cannot use the expression tree in a context which has only execution
             // permissions.  We should endeavour to translate constants into 
             // new constants which have public types.
-            TypeInfo typeInfo = t.GetTypeInfo();
-            if (typeInfo.IsGenericType && typeInfo.GetGenericTypeDefinition().GetTypeInfo().ImplementedInterfaces.Contains(typeof(IGrouping<,>)))
+            if (t.IsGenericType && t.GetGenericTypeDefinition().GetInterfaces().Contains(typeof(IGrouping<,>)))
```

Comment:
Do we have coverage for this?  This involves changes in API calls, so just verifying.